packages/xstate-wallet/src/store/store.ts

```diff
@@ -528,6 +562,51 @@ export class Store {
         return currentBudget;
       }
     );
+
+  /**
+   *  START of wallet 2.0
+   */
+
+  async pushMessage2(signedStates: SignedState[] | undefined, objectives: Objective[] | undefined) {
+    if (objectives) await Promise.all(objectives.map(_.bind(this.addRichObjective, this)));
```

Comment:
Either one works. Originally, `_.bind` was not used but was introduced when we saw runtime errors.

Reply:
`objectives.map(this.addRichObjective)` doesn't work because sometimes (always?), when generating the AST* for `pushMessage2`, `this.addRichObjective` holds a reference to `undefined`. 

`objectives.map(o => this.addRichObjective(o)` works because `this.addRichObjective` is not executed until the function passed to `map` is called. At that point in time, `this.addRichObjectives` holds a reference to a function.

In `server-wallet`, we do this to replace `this.addRichObjective` with `this.addRichObjective.bind(this)` when the class is constructed.

`_.bind` is a third pattern.

Is it worth being consistent with how we address this issue

\*I don't have a definite understanding of how this all works, but I think that's "accurate enough".

Reply:
> In server-wallet, we do this to replace this.addRichObjective with this.addRichObjective.bind(this) when the class is constructed.

Actually, I did a quick search and I'm no longer sure that we do this. We definitely did this at some point in the past.

Reply:
My understanding of the reason that `this` must be bound to a function is described [here](https://stackoverflow.com/a/20279485/13280122)

Reply:
According to that link, I think that
```suggestion
    if (objectives) await Promise.all(objectives.map(this.addRichObjective, this));
```
would also work?

I don't know if it matters at all, but it seems a bit "off" to bind `this.addRichObjective` to `this` every time L571 is called. My gut says that binding to `this` should be done once to create the properly bound function, and that function is passed as a callback. But, perhaps this doesn't matter at all.